web/src/app/users/UserContactMethodDeleteDialog.js

```diff
@@ -2,41 +2,41 @@ import React from 'react'
 import p from 'prop-types'
 import gql from 'graphql-tag'
 import FormDialog from '../dialogs/FormDialog'
-import { Mutation } from 'react-apollo'
+import { useMutation } from '@apollo/react-hooks'
 import { nonFieldErrors } from '../util/errutil'
 
 const mutation = gql`
   mutation($id: ID!) {
     deleteAll(input: [{ id: $id, type: contactMethod }])
   }
 `
-export default class UserContactMethodDeleteDialog extends React.PureComponent {
-  static propTypes = {
-    contactMethodID: p.string.isRequired,
-    onClose: p.func.isRequired, // passed to FormDialog
-  }
+function UserContactMethodDeleteDialog(props) {
+  const { contactMethodID, ...rest } = props
 
-  render() {
-    return (
-      <Mutation mutation={mutation} onCompleted={this.props.onClose}>
-        {(commit, status) => this.renderDialog(commit, status)}
-      </Mutation>
-    )
-  }
+  const [deleteUserMethod, deleteMethodStatus] = useMutation(mutation, {
```

Comment:
`[deleteCM, deleteCMStatus]` might be better names. When I read "deleteUserMethod" I could see it interpreted it as a method that will delete a user when called

Reply:
makes sense, all fixed!